File: GoogleSharp/GoogleSharp/src/core/structures/Document.cs
namespace GoogleSharp.Src.Core.Structures {
    public class Document {
        public Document (string id, string path) {
            Id = id;
            Path = path;
        }

        public string Id { get; set; }
        public string Path { get; set; }

        public override bool Equals (object obj) {
            if (obj is Document other)
                return Id == other.Id;
            return false;
        }

        public override int GetHashCode () {
            return Id.GetHashCode ();
        }

        public override string ToString () {
            return $"Document({Id})";
        }
    }
}

File: google/main/src/core/structures/Document.java
package main.src.core.structures;


public class Document extends Entity{

    private String path;

    public Document(String id, String path) {
        super(id);
        this.path = path;
    }

    public String getPath() {
        return this.path;
    }

    public void setPath(String newPath) {
        this.path = newPath;
    }

    @Override
    public String toString() {
        return "Document(" + this.id + ")";
    }
}